packages/next/src/storage.ts

```diff
@@ -21,9 +23,15 @@ export default class NextStorage implements Storage {
 
   removeItem(key: StorageKey) {
     this.request.session[key] = undefined;
+    this.sessionChanged = true;
   }
 
   async save() {
+    if (!this.sessionChanged) {
+      return;
+    }
+
     await this.request.session.save();
+    this.sessionChanged = false;
```

Comment:
What if the session changed to `true` when saving the session, in a high concurrency situation?